specs/AnimatedVisualPlayer Spec.md

```diff
@@ -0,0 +1,145 @@
+control.
+`AnimatedVisualPlayer` can play any
+[IAnimatedVisual](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.IAnimatedVisual)
+implementation,
+but the typical way to get an `IAnimatedVisual` implementation is to use
+the LottieGen tool to generate code from a JSON file that was created from Adobe AfterEffects.
+
+[More information on Lottie and AnimatedVisualPlayer](https://docs.microsoft.com/en-us/windows/communitytoolkit/animations/lottie)
+
+There are some performance issues with the current implementation of LottieGen and AnimatedVisualPlayer: 
+1)	Complex Lottie animations can instantiate a lot of composition animations 
+which may impact performance and overall battery life. 
+2)	Lottie files that have a lot of complex animations even when they are paused
+have some cost to performance. 
+The Idle (paused) Lottie animations use the same amount of CPU resources as
+when they are playing the actual animation.
+
+The new APIs here allow `AnimatedVisualPlayer`,
+[AnimatedIcon](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.AnimatedIcon),
+and the interface generated by LottieGen to be updated to help reduce the number of composition animations created
+when the animation is idle or paused to help improve performance when a developer uses
+Lottie Animations with `AnimatedVisualPlayer`.
+
+This spec will focus on changes for LottieGen and `AnimatedVisualPlayer`. 
+
+This will require 3 changes: 
+1. Create a new `IAnimatedVisual2` interface which adds to
+[IAnimatedVisual](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.IAnimatedVisual)
+the ability to destroy and instantiate animations.
+2. Create a new `IAnimatedVisualSource3` interface that extends
+[IAnimatedVisualSource](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.IAnimatedVisualSource)
+with the ability to return the new `IAnimatedVisual2`
+3.	Add a `CacheMode` property to `AnimatedVisualPlayer` that will control
+when animations should be destroyed and when they should be instantiated.
+This will be implemented using the previous new interfaces.
+
+# API Pages
+
+## AnimatedVisualPlayer.CacheMode property
+
+Gets or sets a value that determines how animations are cached. Defaults to 'Always'.
+
+The two modes for the property will specify the behavior the AnimatedVisualPlayer will have
+when the player is idle (when `PlayAsync` is not active).
+
+|Mode| Behavior|
+|---- | -------|
+|Always| `AnimatedVisualPlayer` will pre-create animations even before `PlayAsync()` is called,
+and not destroy any when player is idle. |
+|None | `AnimatedVisualPlayer` will not create animations until `PlayAsync()` is called,
+and will destroy them when it completes. |
+
+If you have an animation that will start on click, you may want to set `CacheMode` to `Always`
+when the mouse enters this control, 
+and back to `None` when mouse leaves to ensure that the animation starts immediately on click,
+but don't use up resources otherwise.
```

Comment:
"Always/None" appears to be mismatched. The opposite of "Always" is "Never". But "Never" doesn't seem to be right, since that suggests that we never create the animation at all.

Always/WhenPlaying?

But calling this "CacheMode" suggests that when the animation is not playing (and we are in "None" mode), then the entire animation is discarded, even the static image. If the CacheMode is "None", and the animation is paused or stopped, does it render nothing? Or does it render the final static frame?

My guess is that the static image is still rendered. What this controls is whether we keep the resources related to animations (non-presented frames).

Maybe this is really the AnimationCacheMode (Always, WhenPlaying). I'm still not a big fan of my own name.

Reply:
I agree with "Never" -> "WhenPlaying", and it seems that it is already called "AnimationCacheMode" in my draft PR with code, it just not updated in the specs yet